controllers/apps/component_controller.go

```diff
@@ -168,6 +174,8 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			&componentConfigurationTransformer{Client: r.Client},
 			// handle restore before workloads transform
 			&componentRestoreTransformer{Client: r.Client},
+			// handle upgrade from the legacy RSM API to the InstanceSet API
+			&upgradeTransformer{},
```

Comment:
It might be more semantically appropriate to rename it to componentWorkloadUpgraderTransformer, as the current name could easily be misunderstood as an upgrade of the component itself.

Reply:
fixed by [e1e6052](https://github.com/apecloud/kubeblocks/pull/7087/commits/e1e6052ca89391e27ea523370114bb7e3d780cbc)